src/OrbitBase/include/OrbitBase/FutureHelpers.h

```diff
@@ -73,6 +73,82 @@ template <typename T>
   return promise->GetFuture();
 }
 
+template <typename T>
+[[nodiscard]] inline Future<ErrorMessageOr<T>> UnwrapFuture(
+    const Future<ErrorMessageOr<Future<T>>>& outer_future) {
+  orbit_base::Promise<ErrorMessageOr<T>> promise{};
+  auto future = promise.GetFuture();
+
+  auto outer_continuation =
+      [promise = std::move(promise)](const ErrorMessageOr<Future<T>>& outer_result) mutable {
+        if (outer_result.has_error()) {
+          promise.SetResult(outcome::failure(outer_result.error()));
```

Comment:
do you need outcome::failure here? I think you can drop it.

Reply:
Unfortunately not. The implicit constructor of the error case is only available if it's unambiguous.

So for example it won't be available for the type `ErrorMessageOr<ErrorMessage>` or when the `T` in `ErrorMessageOr<T>` has a templated constructor. So to make this universally work, we need outcome::failure.

I have another commit (not yet put out) which will add this to other parts of the code as well.

Reply:
I'm going to resolve this.